projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -198,6 +202,33 @@ def ingest_payload(
             invocations belong to same collection. Defaults to "data_job_name|OpID",
             meaning all method invocations from a data job run will belong to the
             same collection.
+
+        :return: [Optional] IngestionResult, containing plugin-defined
+        information about the result from the ingestion operation. This result
+        could be used for post-ingestion processing operations like sending
+        telemetry data about the ingested payload.
+        """
+        pass
+
+    def post_ingest_process(
+        self,
+        payload: Optional[List[dict]] = None,
+        ingestion_result: Optional[IngestionResult] = None,
+    ) -> None:
+        """
+        Do post-ingestion processing of the ingestion payload
+        or result data from the ingestion operation.
```

Comment:
What if ingestion raised an exception, is the error a specific outcome of the ingestion operation? If so, there would be an use case that telemetry would use this capability to monitor ingestion errors also? 

Or perhaps another capability in addition to this one, trying to understand should we have one or more hooks for both use cases

Reply:
Did some refactoring. Now, it should be able to catch all exceptions and pass them as a list to the post-ingestion hook.